lib/node_modules/@stdlib/blas/ext/base/zfill/lib/ndarray.js

```diff
@@ -86,49 +84,50 @@ function zfill( N, alpha, x, stride, offset ) {
 	// Reinterpret the complex input array as a real-valued array:
 	view = reinterpret( x, 0 );
 
-	// Use loop unrolling if the stride is equal to `1`...
-	if ( stride === 1 ) {
+	// Adjust the stride and offset according to real-valued array:
+	ix = offsetX * 2;
+	strideX *= 2;
+
+	// Use loop unrolling if the stride is equal to `2`...
+	if ( strideX === 2 ) {
 		m = N % M;
```

Comment:
@kgryte this should be `m = N % (M/2)`, same change in the loop below.

To avoid this calculation for each iteration in the loop below, we will need to introduce a new variable in the case `HALF_M = M/2` or for the earlier version.

Reply:
@headlessNode Okay. We could also adjust `N` here. Regardless, this now seems more trouble than its worth. Let's just stick with `M = 8` with `ix += M * 2`.

Reply:
Yes, I was thinking the same.